src/lib/device_tree/src/lib.rs

```diff
@@ -142,24 +136,44 @@ fn align4(x: usize) -> usize {
     (x + 3) & !3
 }
 
+// Reads the header of the device tree. 
+fn read_fdt_header(drv: &dyn Driver) -> Result<FdtHeader> {
+    // 10 fields, each field 4 bytes.
+    const HEADER_SIZE: usize = 10 * 4;
+
+    let mut data = [0; HEADER_SIZE];
+    let size = drv.pread(&mut data, 0)?;
+    if size != HEADER_SIZE {
+        return Err("not enough data to read device tree header");
```

Comment:
if would be nice at this point to tell us just how much data you were able to read.

Reply:
I would love that as well, but because the error is of type `&'static str`, I believe we can only use string literals (there is no way I know of to add custom information like number of bytes here). :(